fbench/src/fbench/client.cpp

```diff
@@ -220,8 +224,15 @@ Client::run()
                 strcat(_linebuf, _args->_queryStringToAppend.c_str());
             }
             int cLen = _args->_usePostMode ? urlSource.nextContent() : 0;
+            auto content = urlSource.content();
+            if (_args->_usePostMode && _args->_base64Decode) {
+                auto base64_decoded = Base64::decode(std::string(content, cLen));
```

Comment:
You should not wrap the input as an std::string. That will make a copy of the content. Just use the the decode(const char *, int length) method.